src/formatDate.js

```diff
@@ -8,7 +8,38 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const [fromSeparator] = fromFormat.splice(-1, 1);
+  const [toSeparator] = toFormat.splice(-1, 1);
+
+  const dateElements = date.split(fromSeparator);
+
+  const convertYear = (year, format) => {
+    if (format === 'YYYY') {
+      return year.length === 2 ? (year < 30 ? '20' + year : '19' + year) : year;
+    }
+
+    if (format === 'YY') {
+      return year.slice(-2);
+    }
+
+    return year;
```

Comment:
The `convertYear` function assumes that the year will be at the end of the `fromFormat`, which may not always be the case. It would be more robust to look up the year using the format strings ('YYYY' or 'YY') from the `dateMap`.